src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int DEFAULT_MULTIPLIER = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table[getIndexByKey(key)] == null) {
+            putOnEmpty(key, value);
+        } else {
+            insertValue(key, value);
+        }
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndexByKey(key)];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.nextNode;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndexByKey(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private void putOnEmpty(K key, V value) {
+        Node<K, V> newNode = new Node<>(key, value, null);
+        table[getIndexByKey(key)] = newNode;
+        size++;
+    }
+
+    private void insertValue(K key, V value) {
+        Node<K, V> currentNode = table[getIndexByKey(key)];
```

Comment:
`table[getIndexByKey(key)]` this code repeats in public and both private methods
maybe we can pass this value as method parameter?